src/OpenIDConnectClient.php

```diff
@@ -307,6 +307,11 @@ public function authenticate() {
                 throw new OpenIDConnectClientException('Got response: ' . $token_json->error);
             }
 
+	        // Sometime getState() return an empty string
+            // and the authentication process fail
+            if ($this->getState() == "")
+                $this->setState($_REQUEST['state']);
```

Comment:
This happens when the session does not contain the state.

Does this introduce security issues? 
It looks to me that it is now possible to reload the page with the `?code=<somecode>&state=<some state>`. Instead that it was normally not possible because the state was unset.

Maybe the specs says something about this?

Reply:
We should for sure not do this - if the session does not hold the necessary values there is something broken.

Same for the nonce below